### ISSUE TITLE
Compiler: Assignment Type and Dereference Level Checks

### ISSUE THREAD

**Issue by Mitch-Siegel** [Mitch-Siegel/customSystem#23]
Similarly to the problems in issue #22, now that the compiler is becoming usable, it's also becoming dangerous. An operand type consistency checker for the IR will help easily weed out potential problems in source code. Preventing funny assignments by requiring some sort of c-style casting system is the ultimate goal, but for now a simple warning system will be helpful enough. I want to lay the groundwork for this before starting to work on a type system.

**Comment by Mitch-Siegel** [comment]
Type/dereference level checks implemented in commit 3fd52bb1d42cde763dd5882f5f7cb692d801e330. This is done through a scanning pass over the IR after the linearization step. Warnings are emitted, including information about the AST corresponding to the problematic line of TAC. Additionally, AST nodes now track the line and column of source code they came from, and this information is emitted with warnings as well.

This system will need to be revisited when types are implemented, but that is a significant enough change that I think it warrants a separate issue.
